job-server/src/main/scala/spark/jobserver/WebApi.scala

```diff
@@ -345,7 +340,9 @@ class WebApi(system: ActorSystem,
       get { ctx =>
         (supervisor ? ListContexts).mapTo[Seq[String]]
           .map { contexts =>
-            ctx.complete(removeProxyUserPrefix(authInfo, contexts)) }
+            ctx.complete(SparkJobUtils.removeProxyUserPrefix(
+              authInfo.toString, contexts,
```

Comment:
If authentication is not used then authInfo is new AuthInfo(new User("anonymous")). I have changed that to be lazy evaluated parameter of SparjJobUtils.removeProxyUserPrefix